src/backend/joanie/payment/models.py

```diff
@@ -347,12 +347,12 @@ def save(self, *args, **kwargs):
 class CreditCardManager(models.Manager):
     """Custom manager for `CreditCard` model"""
 
-    def get_card_for_owner(self, pk, owner_id):
+    def get_card_for_owner(self, pk, owner_id, payment_provider):
```

Comment:
IMO the `payment_provider` should not be an argument. You should retrieve it inside this method.
```
payment_backend = get_payment_backend()
```
then
```
payment_provider=payment_backend.name
```